controller/monitor/environment_check_monitor.go

```diff
@@ -0,0 +1,558 @@
+		return data, errors.Wrap(err, "failed to copy collected data")
+	}
+
+	return data, nil
+}
+
+func (m *EnvironmentCheckMonitor) run(value interface{}) error {
+	node, err := m.ds.GetNode(m.nodeName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get longhorn node %v", m.nodeName)
+	}
+
+	collectedData := m.collectEnvironmentCheckData(node)
+	if !reflect.DeepEqual(m.collectedData, collectedData) {
+		func() {
+			m.collectedDataLock.Lock()
+			defer m.collectedDataLock.Unlock()
+			m.collectedData = collectedData
+		}()
+
+		key := node.Namespace + "/" + m.nodeName
+		m.syncCallback(key)
+	}
+
+	return nil
+}
+
+func (m *EnvironmentCheckMonitor) collectEnvironmentCheckData(node *longhorn.Node) *CollectedEnvironmentCheckInfo {
+	kubeNode, err := m.ds.GetKubernetesNodeRO(node.Name)
+	if err != nil {
+		return &CollectedEnvironmentCheckInfo{
+			conditions: []longhorn.Condition{},
+		}
+	}
+
+	return m.environmentCheck(kubeNode)
+}
+
+func (m *EnvironmentCheckMonitor) environmentCheck(kubeNode *corev1.Node) *CollectedEnvironmentCheckInfo {
+	collectedData := &CollectedEnvironmentCheckInfo{
+		conditions: []longhorn.Condition{},
+	}
+
+	// Need to find the better way to check if various kernel versions are supported
+	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
+	m.syncPackagesInstalled(kubeNode, namespaces, collectedData)
+	m.syncMultipathd(namespaces, collectedData)
+	m.syncNFSClientVersion(kubeNode, collectedData)
+
+	isV2DataEngine, err := m.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
+	if err != nil {
+		m.logger.WithError(err).Debug("Failed to fetch v2-data-engine setting")
+		isV2DataEngine = false
+	}
+
+	m.checkKernelModulesLoaded(kubeNode, isV2DataEngine, collectedData)
+
+	if isV2DataEngine {
+		m.checkHugePages(kubeNode, collectedData)
+	}
+
+	return collectedData
+}
+
+func (m *EnvironmentCheckMonitor) syncPackagesInstalled(kubeNode *corev1.Node, namespaces []lhtypes.Namespace, collectedData *CollectedEnvironmentCheckInfo) {
+	osImage := strings.ToLower(kubeNode.Status.NodeInfo.OSImage)
+	queryPackagesCmd := ""
+	options := []string{}
+	packages := []string{}
+	pipeFlag := false
+
+	switch {
+	case strings.Contains(osImage, "talos"):
+		m.syncPackagesInstalledTalosLinux(namespaces, collectedData)
+		return
+	case strings.Contains(osImage, "ubuntu"):
+		fallthrough
+	case strings.Contains(osImage, "debian"):
+		queryPackagesCmd = "dpkg"
+		options = append(options, "-l")
+		packages = append(packages, "nfs-common", "open-iscsi", "cryptsetup", "dmsetup")
+		pipeFlag = true
+	case strings.Contains(osImage, "centos"):
+		fallthrough
+	case strings.Contains(osImage, "fedora"):
+		fallthrough
+	case strings.Contains(osImage, "red hat"):
+		fallthrough
+	case strings.Contains(osImage, "rocky"):
+		fallthrough
+	case strings.Contains(osImage, "ol"):
+		queryPackagesCmd = "rpm"
```

Comment:
Do we still use `rpm` command  to check if pkgs are installed?

Reply:
The purpose of the PR is not to remove `rpm`. It just moves the checks to a monitor and prevents frequent checks. Also fix some bugs.